etl/snapshot.py

```diff
@@ -111,26 +66,18 @@ def metadata_path(self) -> Path:
         else:
             return Path(f"{paths.SNAPSHOTS_DIR / self.uri}.dvc")
 
-    def _cached_dvc_repo(self) -> Any:
-        """Return DVC repo and cache it. Backported steps do not use cached repository."""
-        if "backport" in str(self.metadata_path):
-            # The repo has to be created again to pick unignored files
-            return get_dvc(use_cache=False)
-        else:
-            return get_dvc(use_cache=True)
-
     def _download_dvc_file(self, md5: str) -> None:
-        """Download file from DVC remote to self.path."""
+        """Download file from remote to self.path."""
         self.path.parent.mkdir(exist_ok=True, parents=True)
         if self.metadata.is_public:
-            https_url = DVC_CONFIG['remote "public-read"']["url"]
-            download_url = f"{https_url}/{md5[:2]}/{md5[2:]}"
+            download_url = f"{config.R2_SNAPSHOTS_PUBLIC_READ}/{md5[:2]}/{md5[2:]}"
             files.download(download_url, str(self.path), progress_bar_min_bytes=2**100)
         else:
-            s3_url = DVC_CONFIG['remote "private"']["url"]
-            download_url = f"{s3_url}/{md5[:2]}/{md5[2:]}"
+            download_url = f"{config.R2_SNAPSHOTS_PRIVATE}/{md5[:2]}/{md5[2:]}"
             s3_utils.download(download_url, str(self.path))
 
+        assert checksum_file(self.path) == md5, f"Checksum mismatch for {self.path}"
```

Comment:
I wonder if it's appropriate to leave the file in place if the checksum is wrong. Should we rename it to `.bad` or something? The worry is that we forget or rerun and it doesn't get checked again.

Open question, I'm really not sure.

Reply:
Good idea, better to remove it. It's worth saying that this should NEVER happen and I was surprised to find one snapshot and two walden datasets that failed this assert. This was due to my recent change in md5 calculation that fixed newlines compatiblity between Windows/Linux in csv files. I fixed thoes files and we should hopefully never again run into this error.